### PR TITLE
PEX: Provide ParseEnvelope to correctly parse PEX VP envelopes

### PR DESCRIPTION
**Pull request by reinkrul**
Excerpt from https://github.com/nuts-foundation/nuts-node/pull/2518

### CODE REVIEW

File: vcr/pe/util.go [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+package pe
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/lestrrat-go/jwx/v2/jwt"
+	"github.com/nuts-foundation/go-did/vc"
+)
+
+type Envelope struct {
```
Comment (woutslakhorst) [inline vcr/pe/util.go:28]: godoc

File: vcr/pe/util.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+	Interface     interface{}
+	Presentations []vc.VerifiablePresentation
```
Comment (woutslakhorst) [inline vcr/pe/util.go:30]: Do these need to be public

File: vcr/pe/presentation_submission.go [hunk 3]
```diff
@@ -260,29 +260,15 @@ func resolveCredential(path []string, mapping InputDescriptorMappingObject, valu
 // The credentials will be returned as map with the InputDescriptor.Id as key.
 // The Presentation Definitions are passed in the envelope, as specified by the PEX specification.
 // It assumes credentials of the presentations only map in 1 way to the input descriptors.
-func (s PresentationSubmission) Validate(envelope interface{}, definition PresentationDefinition) (map[string]vc.VerifiableCredential, error) {
-	actualCredentials, err := s.Resolve(envelope)
+func (s PresentationSubmission) Validate(envelope Envelope, definition PresentationDefinition) (map[string]vc.VerifiableCredential, error) {
+	actualCredentials, err := s.Resolve(envelope.Interface)
```
Comment (woutslakhorst) [inline vcr/pe/presentation_submission.go:264]: Can't the Resolve also work on the Envelope? The switch at the beginning might be removed.

**Review by woutslakhorst — COMMENTED** [review]
Expected more code removal due to envelope usage

File: vcr/pe/util.go (hunk 2 above)
Comment (reinkrul) [reply vcr/pe/util.go:30]: Only "Presentations"

**Pull request merged**